### PR TITLE
OCPBUGS-32257: Restrict image registry overrides to control plane components

### PR DESCRIPTION
**Pull request by muraee**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #[OCPBUGS-32257](https://issues.redhat.com/browse/OCPBUGS-32257)

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: support/releaseinfo/registry_image_content_policies.go [hunk 1]
```diff
@@ -17,6 +18,10 @@ type ProviderWithOpenShiftImageRegistryOverridesDecorator struct {
 	lock sync.Mutex
 }
 
+func (p *ProviderWithOpenShiftImageRegistryOverridesDecorator) LookupIgnoreOverrides(ctx context.Context, image string, pullSecret []byte) (*ReleaseImage, error) {
```
[the remark this answers is not in the document]
Comment (muraee) [reply support/releaseinfo/registry_image_content_policies.go]: updated

**Review by stevekuznetsov — COMMENTED** [review]
/lgtm
/assign @csrwng

**Pull request merged**